src/main/kotlin/com/hibob/academy/resource/PetController.kt

```diff
@@ -69,19 +73,37 @@ class PetController (private val petService: PetService) {
     @Path("/{companyId}/{petId}/owner")
     fun getOwnerByPetId(@PathParam("petId") petId: Long, @PathParam("companyId") companyId: Long): Response {
         val owner = petService.getOwnerByPetId(petId, companyId)
-           return Response.ok(owner).build()
+        return Response.ok(owner).build()
     }
 
     @GET
     @Path("/company/{companyId}/owner/{ownerId}")
-    fun getOwnerPets(@PathParam("ownerId") ownerId: Long, @PathParam("companyId") companyId: Long): Response {
-        val ownerPets = petService.getPetsByOwnerId(ownerId, companyId)
-        return Response.ok(ownerPets).build()
+    fun getPetsByOwnerId(@PathParam("ownerId") ownerId: Long, @PathParam("companyId") companyId: Long): Response {
+        val pets = petService.getPetsByOwnerId(ownerId, companyId)
+        return Response.ok(pets).build()
     }
 
     @GET
     @Path("/count-by-type/company/{companyId}")
     fun countPetsByType(@PathParam("companyId") companyId: Long): Response {
         return Response.ok(petService.countPetsByType(companyId)).build()
     }
+
+    @PUT
+    @Path("/adopt-multiple-pets/{companyId}/{ownerId}")
+    fun adoptMultiplePets(
+        @PathParam("ownerId") ownerId: Long,
+        @PathParam("companyId") companyId: Long,
+        petList: List<Long>
+    ): Response {
+        petService.adoptMultiplePets(ownerId, companyId, petList)
+        return Response.ok().build()
+    }
+
+    @PUT
```

Comment:
This is creation and not edit method so it should be `POST` instead of `PUT`

Reply:
fixed